compiler/src/iree/compiler/Codegen/LLVMCPU/TileSizeSelection.h

```diff
@@ -0,0 +1,191 @@
+  }
+
+  /// Returns the tiling level for cache reduction dimensions.
+  unsigned getCacheReductionLevel() {
+    return getActualLevel(TilingLevel::CacheReductionTiles);
+  }
+
+  /// Returns the tiling level for vector parallel dimensions.
+  unsigned getVectorParallelLevel() {
+    return getActualLevel(TilingLevel::VectorParallelTiles);
+  }
+
+  /// Returns the tiling level for vector parallel dimensions.
+  unsigned getVectorReductionLevel() {
+    return getActualLevel(TilingLevel::VectorReductionTiles);
+  }
+
+  /// Returns all the tile sizes of all the levels of the configuration.
+  TileSizesListType getTileSizes() { return loweringConfig.getTileSizeVals(); }
+
+  /// Returns the distribution tile sizes of the configuration.
+  SmallVector<int64_t> getDistributionTileSizes() {
+    return loweringConfig.getTileSizeVals(
+        getActualLevel(TilingLevel::DistributionTiles));
+  }
+
+  SmallVector<int64_t> getCacheReductionSizes() {
+    return loweringConfig.getTileSizeVals(getCacheReductionLevel());
+  }
+
+  SmallVector<int64_t> getVectorParallelSizes() {
+    return loweringConfig.getTileSizeVals(getVectorParallelLevel());
+  }
+
+  SmallVector<int64_t> getVectorReductionSizes() {
+    return loweringConfig.getTileSizeVals(getVectorReductionLevel());
+  }
+
+  /// Returns the tile sizes of all the vector dimensions, including parallel
+  /// and reduction dimensions.
+  SmallVector<int64_t> getVectorTileSizes() {
+    unsigned numDims = getNumDimensions();
+    SmallVector<int64_t> vectorSizes(numDims);
+    SmallVector<int64_t> parallelSizes = getVectorParallelSizes();
+    SmallVector<int64_t> reductionSizes = getVectorReductionSizes();
+    for (int i = 0; i < numDims; ++i) {
+      vectorSizes[i] =
+          parallelSizes[i] != 0 ? parallelSizes[i] : reductionSizes[i];
+    }
+
+    return vectorSizes;
+  }
+
+  /// Returns a list with the tiling levels that can be fused for this
+  /// configuration.
+  SmallVector<int64_t> getFusableLevels() {
+    switch (getNumTilingLevels()) {
+      case 0:
+        return {};
+      case 1:
+        // Only distribution level.
+        return {0};
+      case 3:
+        // Distribution + vector parallel levels.
+        return {0, 1};
+      case 5:
+        // Distribution + cache parallel levels.
+        return {0, 1};
+      default:
+        llvm_unreachable("Unexpected number of tiling levels");
+    }
+  }
+
+  // TODO(dcaballe): Revisit if these features are ever used.
+  ArrayAttr getTileInterchange() { return loweringConfig.getTileInterchange(); }
+  SmallVector<int64_t> getTileInterchangeSizes(unsigned level) {
+    return loweringConfig.getTileInterchangeVals(level);
+  }
+  SmallVector<int64_t> getNativeVectorSizes() {
+    return loweringConfig.getNativeVectorSizeVals();
+  }
+
+ private:
+  /// Internal representation for all the supported tiling levels. All or just
+  /// a subset of them may be available in a valid configuration.
+  enum class TilingLevel {
```

Comment:
We prefer C++ casting (i.e., static_cast). It is fine to use `enum` to me because they are scoped to private members. Then we don't need casting at all.